R/estimateAlpha.R

```diff
@@ -0,0 +1,248 @@
+        index <- gsub("_richness", "", index)
+        FUN <- .estimate_richness
+    } else {
+        stop("'index' is coresponding to none of the alpha diversity measures.",
+             call. = FALSE)
+    }
+    
+    if (rarify) {
+        .alpha_rarefaction(x, nrounds = nrounds, seed = seed,
+                           args.sub = list(assay.type=assay.type,
+                                           min_size=rarefaction_depth,
+                                           verbose=FALSE),
+                           FUN=FUN,
+                           args.fun=list(index=index, assay.type="subsampled"),
+                           ...,
+                           name=name)
+    } else {
+        suppressWarnings(do.call(FUN, args = c(list(x, assay.type=assay.type,
+                                                  assay_name=assay_name,
+                                                  index=index, name=name),
+                                                  list(...))))
+    }
+
+}
+
+## Helper functions
+
+.get_indices <- function(measure) {
+    switch(measure,
+           "diversity" = c("coverage_diversity", "coverage",
+                           "faith_diversity", "faith",
+                           "fisher_diversity", "fisher",
+                           "gini_simpson_diversity", "gini_simpson",
+                           "inverse_simpson_diversity", "inverse_simpson",
+                           "log_modulo_skewness_diversity", "log_modulo_skewness",
+                           "shannon_diversity", "shannon"),
+           "dominance" = c("absolute_dominance", "absolute",
+                           "dbp_dominance", "dbp",
+                           "core_abundance_dominance", "core_abundance",
+                           "gini_dominance", "gini", 
+                           "dmn_dominance", "dmn",
+                           "relative_dominance", "relative",
+                           "simpson_lambda_dominance", "simpson_lambda"),
+           "evenness" = c("camargo_evenness", "camargo",
+                          "pielou_evenness", "pielou",
+                          "simpson_evenness",
+                          "evar_evenness", "evar",
+                          "bulla_evenness", "bulla"),
+           "richness" = c("ace_richness", "ace",
+                          "chao1_richness", "chao1",
+                          "hill_richness", "hill",
+                          "observed_richness", "observed"))
+}
+
+.alpha_rarefaction <- function(x,
+                               nrounds=1L,
+                               seed=123,
+                               args.sub=list(assay.type="counts",
+                                             min_size=min(colSums(assay(x, "counts")),
+                                                          na.rm = TRUE),
+                                             verbose=FALSE),
+                               FUN=estimateDiversity,
```

Comment:
should this rather be estimateAlpha?

Reply:
not actually I should change it to .estimate_diversity